addons/escoria-core/game/core-scripts/esc/esc_object_manager.gd

```diff
@@ -15,96 +15,255 @@ const RESERVED_OBJECTS = [
 	CAMERA
 ]
 
+# Separator to use for each room's entry in the objects dictionary.
+const SEPARATOR = "!!!!"
 
-# The hash of registered objects (the global id is the key)
-var objects: Dictionary = {}
+# Dictionary key for reserved objects.
+const RESERVED_KEY = "reserved"
+
+# The hash of registered objects (organized by room, with the object's global id 
+# serving as the object's key).
+#
+# Example structure:
+#
+#	{ 
+#		"reserved":
+#			{
+#				"_camera": camera
+#			},
+#		"room1!!!!<instance_id>":
```

Comment:
I'd go with an array of classes here that have properties like `room`, `instance id`, `objects`-array instead of working with strings and separators.

Reply:
Can do, but it makes lookups a bit less efficient since we now have to disambiguate by going one level deeper into the structure. Are we ok with that? I don't think we'd need `room` as a property inside since all we'd need is the room name which would be the key.